test/route-define-test.js

```diff
@@ -61,14 +61,16 @@ if (typeof steal !== 'undefined') {
 		canRoute.data = appState;
 
 		canRoute._onStartComplete = function () {
-			equal(mockRoute.hash.value, "cat/5&section=home", "same URL");
-			equal(appState.get("type"), "cat", "hash populates the appState");
-			equal(appState.get("id"), "5", "hash populates the appState");
-			equal(appState.get("section"), "home", "appState keeps its properties");
-			ok(canRoute.data === appState, "canRoute.data is the same as appState");
-
-			mockRoute.stop();
-			QUnit.start();
+			setTimeout(function(){
+				equal(mockRoute.hash.value, "cat/5&section=home", "same URL");
+				equal(appState.get("type"), "cat", "hash populates the appState");
+				equal(appState.get("id"), "5", "hash populates the appState");
+				equal(appState.get("section"), "home", "appState keeps its properties");
+				ok(canRoute.data === appState, "canRoute.data is the same as appState");
+	
+				mockRoute.stop();
+				QUnit.start();
+			},0.5);
```

Comment:
Can you explain this?  It's __very__ good practice to put inline comments describing any changes that someone would not understand.

Reply:
_If_ this change is needed, it really should be done inside `_onStartComplete`. The purpose of that function is to tell tests when they can do assertions.

That said, I still don't understand why this change is needed. Why is `_onStartComplete` being called before things have settled?